scripts/deno/surface-lighthouse-results.ts

```diff
@@ -0,0 +1,206 @@
+import { Octokit } from "https://cdn.skypack.dev/octokit?dts";
+import type { RestEndpointMethodTypes } from "https://cdn.skypack.dev/@octokit/plugin-rest-endpoint-methods?dts";
+import type { EventPayloadMap } from "https://cdn.skypack.dev/@octokit/webhooks-types?dts";
+import "https://raw.githubusercontent.com/GoogleChrome/lighthouse-ci/v0.8.2/types/assert.d.ts";
+
+/* -- Setup -- */
+
+/** Github token for Authentication */
+const token = Deno.env.get("GITHUB_TOKEN");
+if (!token) throw new Error("Missing GITHUB_TOKEN");
+
+/** Path for workflow event */
+const path = Deno.env.get("GITHUB_EVENT_PATH");
+if (!path) throw new Error("Missing GITHUB_EVENT_PATH");
+
+/**
+ * https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads
+ */
+const payload: EventPayloadMap["push" | "pull_request"] = JSON.parse(
+	Deno.readTextFileSync(path)
+);
+
+const isPullRequestEvent = (
+	payload: EventPayloadMap[keyof EventPayloadMap]
+): payload is EventPayloadMap["pull_request"] =>
+	//@ts-expect-error -- We’re actually checking the type
+	typeof payload?.pull_request?.number === "number";
+
+/**
+ * One of two values depending on the workflow trigger event
+ *
+ * - on a "pull_request" event: the current Issue / PR number
+ * on which to add or update the Lighthouse report
+ *
+ * - on a "push" event: the original issue to track the latest report:
+ * https://github.com/guardian/dotcom-rendering/issues/4584
+ */
+const issue_number = isPullRequestEvent(payload)
+	? // If PullRequestEvent
+	  payload.pull_request.number
+	: // If PushEvent
+	  4584;
```

Comment:
What's this hard-coded number here?

Reply:
I believe its the original issue # https://github.com/guardian/dotcom-rendering/issues/4584, the action updates the original issue when a PR is pushed to main

Reply:
Would be interesting if we could update the repo's README and surface the numbers there instead?

Reply:
We're going to re-visit this (great) idea in a separate Spike